src/pages/matrix.tsx

```diff
@@ -1,62 +1,246 @@
+   * Scrolls the grid to the given input string if found in the matrix,
+   * otherwise shows a no results message
+   *
+   * @param currInput
+   * @returns
+   */
+  const scrollToInput = (currInput: string | undefined) => {
+    if (!currInput) {
+      return;
+    }
+    if (!gridDataMap[currInput.toLowerCase()]) {
+      showNoResults();
+      return;
+    }
+
+    const selected = gridDataMap[currInput.toLowerCase()];
+    if (selected.type === 'row') {
+      gridRef.current.searchRow(selected.index);
+    } else if (selected.type === 'col') {
+      gridRef.current.searchCol(selected.index);
+    }
+  };
+
+  /**
+   * Handles when the user clicks the search button
+   */
+  const handleSubmit = () => {
+    scrollToInput(input);
+  };
+
+  /**
+   * Handls changes to the Color theme dropdown
+   *
+   * @param option
+   */
+  const handleChangeTheme = (option: any) => {
+    if (option) {
+      setColorThemeOption(option);
+    }
+  };
+
+  const numRows = gridData.length;
+  const numColumns = gridData[0].length;
+
+  // Calculate Layout of the Grid:
+
+  const maxCols = styles.maxCols;
+  const maxRows = styles.maxRows;
+
+  const rowHeaderWidth = styles.rowHeaderWidth;
+  const colHeaderHeight = styles.colHeaderHeight;
+  const cellHeight = styles.cellHeight;
+  const cellWidth = styles.cellWidth;
+
+  const widthBufferSpace = 50; // buffer space for keeping everything in viewport
+  const heightBufferSpace = 330; // buffer space for keeping everything in viewport
+  const gridHeight = Math.min(
+    cellHeight * numRows, // can't exceed total grid
+    height - colHeaderHeight - heightBufferSpace, // can't exceed browser height
+    colHeaderHeight + cellHeight * maxRows
+  );
+  const gridWidth = Math.min(
+    cellWidth * numColumns, // can't exceed total grid
+    width - rowHeaderWidth - widthBufferSpace, // can't exceed browser width
+    rowHeaderWidth + cellWidth * maxCols
+  );
+
+  const gridTitle = styles.title;
+  const gridSubtitle = styles.subtitle;
+
+  const legendWidth = gridWidth + rowHeaderWidth;
+  const legendHeight = styles.legendHeight;
+  const legendItemSize = styles.legendBarWidth;
+  const searchItemHeight = 30;
+
+  const toastStyles: CSSProperties = {
+    position: 'absolute',
+    top: 167,
+    left: width / 2 + 230,
+    padding: '9px 12px',
+  };
+
+  const searchBarSelectProps = {
+    autoFocus: true,
+    onInputChange: handleInputChange,
+    onChange: handleChange,
+    options,
+    inputValue: input,
+    value: input ? { value: input, label: input } : undefined,
+    isClearable: Boolean(input),
+    openMenuOnFocus: false,
+    openMenuOnClick: false,
+  };
+
+  return (
+    <div className='matrix-page'>
+      {toastMessage ? (
+        <div className='alert alert-danger' role='alert' style={toastStyles}>
+          {toastMessage}
+        </div>
+      ) : null}
+      <div className='content-container'>
+        <div className='title-container'>
+          <h1>{gridTitle}</h1>
```

Comment:
Assuming you've changed `gridTitle` to be the output of `renderMarkdown`, then this line should be

```jsx
<h1><DisplayValue addClass value={{value: title, isHTML: true}} /></h1>
```